pkg/apis/pingcap/v1alpha1/defaulting/tidbcluster.go

```diff
@@ -82,6 +83,15 @@ func setTidbSpecDefault(tc *v1alpha1.TidbCluster) {
 		tc.Spec.TiDB.MaxFailoverCount = pointer.Int32Ptr(3)
 	}
 
+	if tc.Spec.TiDB.StatefulSetUpdateStrategy == nil {
+		tc.Spec.TiDB.StatefulSetUpdateStrategy = &apps.StatefulSetUpdateStrategy{
+			Type: apps.RollingUpdateStatefulSetStrategyType,
+			RollingUpdate: &apps.RollingUpdateStatefulSetStrategy{
+				Partition: pointer.Int32Ptr(tc.TiDBStsDesiredReplicas()),
```

Comment:
We should not update the `Partition` here, as we will update the `Partition` in our controller logic while this defaulting logic will be called during each sync [here](https://github.com/pingcap/tidb-operator/blob/master/pkg/controller/tidbcluster/tidb_cluster_control.go#L98).

Reply:
What about we add a new function in the `_member_manager.go` to build this field without defaulting here.